dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -149,19 +209,53 @@ private static void setMetadataKeywords(DepositMetadata depositMetadata, Workflo
-                    relatedIdentifier.setIdentifier(aliasUrl);
-                    relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
-                    return relatedIdentifier;
-                }).collect(Collectors.toList());
+            Workflow workflow, WorkflowVersion workflowVersion, String alias,
+            EntryVersionHelper entryVersionHelper, DockstoreWebserviceConfiguration webserviceConfiguration) {
+
+        List<RelatedIdentifier> aliasList = new ArrayList<>();
+
+        // Add the workflow version alias as a related identifier on Zenodo
+        String aliasUrl = dockstoreUrl + "/aliases/workflows/" + alias;
+        RelatedIdentifier aliasRelatedIdentifier = new RelatedIdentifier();
+        aliasRelatedIdentifier.setIdentifier(aliasUrl);
+        aliasRelatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+        aliasList.add(aliasRelatedIdentifier);
+
+        // Add the UI2 link to the workflow to Zenodo as a related identifier
+        // E.g https://dockstore.org/workflows/aliases/10.5281/zenodo.2630727
+        String sourceControl = workflow.getSourceControl().toString(); // e.g. github.com
+        String organization = workflow.getOrganization(); // e.g. DataBiosphere
+        String repository = workflow.getRepository(); // e.g. topmed-workflows
+        String workflowName = workflow.getWorkflowName(); // e.g. UM_variant_caller_wdl
+        String versionOfWorkflow = workflowVersion.getName();
+
+        String sourceControlPathPrefix = sourceControl + "/" + organization + "/" + repository;
+        // Sometimes the workflow name returned is null so skip it if that is the case
+        // this enables a correct UI2 link and TRS URL to be constructed
+        // E.g. https://dockstore.org/workflows/github.com/DataBiosphere/topmed-workflows/UM_variant_caller_wdl:1.32.0
+        String sourceControlPath = StringUtils.isEmpty(workflowName) ? sourceControlPathPrefix : sourceControlPathPrefix + "/" + workflowName;
+        String workflowVersionURL = dockstoreUrl + "/workflows/"  + sourceControlPath + ":" + versionOfWorkflow;
+        RelatedIdentifier relatedIdentifier = new RelatedIdentifier();
+        relatedIdentifier.setIdentifier(workflowVersionURL);
+        relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+        aliasList.add(relatedIdentifier);
+
+        // Add the workflow Task Registry Service (TRS) URL to Zenodo as a related identifier
+        // E.g. https://dockstore.org/api/api/ga4gh/v2/tools/%23workflow%2Fgithub.com%2FDataBiosphere
+        // %2Ftopmed-workflows%2FUM_variant_caller_wdl/versions/1.32.0/PLAIN-WDL/descriptor/topmed_freeze3_calling.wdl
+        String workflowVersionPrimaryDescriptorPath = "#workflow/" + sourceControlPath;
+        String workflowVersionPrimaryDescriptorPathPlainText = ToolsImplCommon.getUrlFromId(webserviceConfiguration,
```

Comment:
Related to the Codacy suggestion, you can avoid a parameter, rather than passing the `webserviceConfiguration` around by just saving the baseURL (which doesn't change) once in the constructor and then calling `ToolsImplCommon.getUrl` instead of `ToolsImplCommon.getUrlFromId`.

Reply:
Done